packages/code-snippet/src/CodeSnippetWidget.tsx

```diff
@@ -359,3 +539,57 @@ export class CodeSnippetWidget extends MetadataWidget {
     );
   }
 }
+
+/**
+ * A content factory interface for code cell content
+ */
+export interface IContentFactory extends Cell.IContentFactory {
+  /**
+   * Create a new code cell widget.
+   */
+  createCodeCell(options: CodeCell.IOptions): CodeCell;
+
+  /**
+   * Create a new markdown cell widget.
+   */
+  createMarkdownCell(options: CellModel.IOptions): MarkdownCellModel;
+}
+
+/**
+ * The default implementation of an `IModelFactory`.
+ */
+export class ModelFactory {
```

Comment:
These two classes look like they were taken from the console package in lab code, I think rather than copying them we should use/extend those classes instead